api-common-java/src/main/java/com/google/api/pathtemplate/ValidationException.java

```diff
@@ -39,6 +39,7 @@
  */
 public class ValidationException extends IllegalArgumentException {
 
+  @FunctionalInterface
```

Comment:
This and above inner-public interface is so weird, probably because we wanted to mimic the same behavior of `java.util.function.Supplier` in Java 7. That being said, I think adding `FunctionalInterface` to them are appropriate because they are exposed on the surface through other public methods. But we may want to consider replacing these `Supplier` with `java.util.function.Supplier` as part of the Java 8 adoption project. CC: @suztomo

Reply:
Thanks for the comments.

I'll do a research on replacing `Supplier` with `java.util.function.Supplier`. Leave it as-is as this is out of scope of this PR.

Reply:
This would break binary compatibility, and so we'd need a major rev bump.

Reply:
> It would break binary compatibility, and so we'd need a major rev bump.

Are you referring to the idea of replacing `Supplier` with `java.util.function.Supplier`? I think it depends on how we approach it, if we do it naively yes it would be a breaking change. We could approach it in steps like adding alternatives first, replacing the ones that are not on the surface first etc.